x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java

```diff
@@ -245,12 +246,25 @@ public class InternalUsers {
         new RoleDescriptor(
             UsernamesField.LAZY_ROLLOVER_ROLE,
             new String[] {},
-            new RoleDescriptor.IndicesPrivileges[] {
-                RoleDescriptor.IndicesPrivileges.builder()
-                    .indices("*")
-                    .privileges(LazyRolloverAction.NAME)
-                    .allowRestrictedIndices(true)
-                    .build() },
+            DataStream.isFailureStoreFeatureFlagEnabled()
```

Comment:
Two TODOs before this PR is fully ready:

1. Think through whether this is a good enough safe-guard -- we end up in a situation where `manage_failure_store` can't get parsed by a node, the node will fail to boot.
2. Check if any other internal users need failure-store privileges.

Reply:
> Think through whether this is a good enough safe-guard -- we end up in a situation where manage_failure_store can't get parsed by a node, the node will fail to boot.

I think this is actually safe -- if the feature flag is set, then the privilege is available on that node and the node will boot. If it isn't enabled, the privilege won't be included and boot will succeed.